### PR TITLE
Apply prefer_final_locals to for loop variables as well

### LINKED ISSUE #1065: prefer_final_locals should handle variable in for-in declaration
```dart
for (var e in list) {}
```

can be

```dart
for (final e in list) {}
```

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #1065

### CODE REVIEW

File: test/rules/prefer_final_locals.dart [hunk 1]
```diff
@@ -2,21 +2,31 @@
...
+
+void fn() {
+  for (var i in [1, 2, 3]) { // LINT
+    print(i);
+  }
+
+  for (final i in [1, 2, 3]) { // OK
+    print(i);
+  }
+}
```
[the remark this answers is not in the document]
Comment (srawlins) [reply test/rules/prefer_final_locals.dart]: Oops, uncovered a bug! Thanks and done.

**Comment by pq** [comment]
@srawlins: looks like this is throwing an exception on the benchmark bot?

https://travis-ci.org/dart-lang/linter/jobs/474712239

File: test/rules/prefer_final_locals.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (bwilkerson) [reply test/rules/prefer_final_locals.dart]: I thought it might.

**Comment by bwilkerson** [comment]
@pq Can we improve the output on the benchmark bot to include at least one of the stack traces from the exceptions? It isn't clear to me why there's an exception, and I'm guessing more information would help.

**Comment by srawlins** [comment]
Great catch, benchmarking tool! This caught cases like `for (a in b) { ... }`, where a is a variable declared outside of the for-each loop. Fixed.

**Comment by pq** [comment]
@bwilkerson:  I think these are not getting propagated by `analyzer`?

> Exception while using a PreferFinalLocals to visit a ForEachStatementImpl in BlockImpl in BlockFunctionBodyImpl in FunctionExpressionImpl in FunctionDeclarationImpl in CompilationUnitImpl

**Comment by bwilkerson** [comment]
@pq The message is being produced in `ExceptionHandlingDelegatingAstVisitor.logException`, but it sends the message (and stack trace) to a logger. Is the code for the benchmark in the linter package? I suspect we just need to update the logger to print out the exception, but I don't know where the logger code is.

**Pull request merged**